Automattic/jetpack
# Backup Plugin: Allow non owner admins to access the Backup dashboard

#### Changes proposed in this Pull Request:

We were checking if the current user is connected but instead we need to check if the site has an owner

See #26100 for a similar approach introduced in Jetpack Search

#### Before

<img width="1680" alt="image" src="https://user-images.githubusercontent.com/746152/189122235-6062f4ad-40c8-4b3f-ab56-84bd8135500a.png">


#### Other information:


#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Launch a JN site with Jetpack Backup on it
- Connect and purchase Backup
- Create a secondary administrator
- Sign in as this other administrator
- Visit the Jetpack Backup page
- Confirm you can see the Backup dashboard
